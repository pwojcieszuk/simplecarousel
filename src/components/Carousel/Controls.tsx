import React from "react";
import styles from "components/Carousel/carousel.module.scss";
import { ControlsProps as Props } from "components/Carousel/logic/types";
import prepareButtons from "components/Carousel/logic/prepareButtons";

const Controls: React.FC<Props> = ({ dispatch, items, step }) => (
  <div className={styles.carouselBtns}>
    <button onClick={() => dispatch({ type: "goToPrevItem" })}>Prev</button>
    {prepareButtons(
      step,
      Object.keys(items).map((i) => parseInt(i))
    ).map((childIndex) => (
      <button
        key={childIndex}
        onClick={() => dispatch({ type: "goToItem", itemIndex: childIndex })}
        id={`${childIndex}`}
      />
    ))}
    <button onClick={() => dispatch({ type: "goToNextItem" })}>Next</button>
  </div>
);

export default Controls;
