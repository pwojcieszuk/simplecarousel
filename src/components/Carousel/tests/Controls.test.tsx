import React from "react";
import { shallow } from "enzyme";
import Controls from "components/Carousel/Controls";

describe("Carousel", () => {
  it("Renders", () => {
    const wrapper = shallow(
      <Controls dispatch={() => true} items={[0]} step={3} currentItem={0} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
