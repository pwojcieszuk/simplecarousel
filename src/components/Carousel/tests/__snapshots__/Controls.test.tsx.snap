// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Renders 1`] = `
<Fragment>
  <div
    className="carouselPrevBtn"
    onClick={[Function]}
  />
  <div
    className="carouselBtns"
  />
  <div
    className="carouselNextBtn"
    onClick={[Function]}
  />
</Fragment>
`;
